src/watts_ui/watts_ui.py

```diff
@@ -0,0 +1,610 @@
+    tolerance = float(str(operation.convergence_criteria.value))
+    convergence_params = str(operation.convergence_params.value).strip('\"')
+    to_sub_params = convert_to_list(operation.to_sub_params)
+    to_main_params = convert_to_list(operation.to_main_params)
+
+    convergence_list = []
+    conv = True
+    while conv:
+
+        # Run the main plugin
+        watts_params, app_result_plugin_1 = run_direct(
+            watts_params, watts_plugins[plugin_1])
+
+        # Extract the values of the last index of the iterating
+        # parameters. This step is necessary because the results
+        # created by WATTS plugins could be list or ndarray with
+        # multiple values (especially for MOOSE based app).
+        watts_params = get_last_value(watts_params, to_sub_params)
+
+        # Run the sub plugin
+        watts_params, app_result_plugin_2 = run_direct(
+            watts_params, watts_plugins[plugin_2])
+
+        # Extract the values of the last index of the iterating
+        # parameters. This step is necessary because the results
+        # created by WATTS plugins could be list or ndarray with
+        # multiple values (especially for MOOSE based app).
+        watts_params = get_last_value(watts_params, to_main_params)
+
+        convergence_list.append(watts_params[convergence_params])
+
+        if len(convergence_list) > 1 and ((convergence_list[-1] - convergence_list[-2])/convergence_list[-1]) < tolerance:
+            conv = False
+            if len(convergence_list) > nmax:
+                conv = False
+
+    watts_params["convergence_list"] = convergence_list
+
+    return (watts_params, app_result_plugin_1, app_result_plugin_2)
+
+
+def run_parametric(watts_params, plugin, wf_level):
+    """Run workflow
+
+    Parameters
+    ----------
+    watts_params
+        Watts params with stored user input parameters
+    plugin
+        Dictionary of plugin
+    wf_level
+        Level of workflow
+
+    Returns
+    -------
+    watts_params
+        Updated WATTS parameters
+    app_result
+        WATTS results
+
+    """
+    operation = wf_level.parametric
+    parametric_name = str(operation.changing_params.value).strip('\"')
+    parametric_list = []
+    app_result = {}
+    for n, val in enumerate(operation.changing_values.value):
+        watts_params[parametric_name] = float(str(val))
+        parametric_list.append(float(str(val)))
+
+        watts_params, app_result_parametric = run_direct(
+            watts_params, watts_plugins[plugin_ID])
+        # Store the results from each individual run to
+        # the 'app_result' dictionary as individual tuple.
+        app_result[f"run_{n}"] = app_result_parametric
+
+    watts_params[f"{parametric_name}_list"] = parametric_list
+    return (watts_params, app_result)
+
+
+# Need to update and get properly from workbench the executable path and the argument
+watts_path = "/Users/zhieejhiaooi/Documents/ANL/watts-devel/watts/src/watts_ui/"
```

Comment:
This needs to be updated. Right now, a model will crash because of this link. We should not ask a user to update this link in the source code.